app/services/create_vote_service.rb

```diff
@@ -0,0 +1,50 @@
+class CreateVoteService
+
+  def initialize(votable,current_user,params)
+    @votable = votable
+    @current_user = current_user
+    @params = params
+  end
+
+  def call
+    if @votable.votes.where(user_id: @current_user.id).present?
+      vote = @votable.votes.where(user_id: @current_user.id).first
+      if vote.value == 1
+        if (@params[:value] == "+1")
```

Comment:
Hmm, I've sorta realized that even though small changes like that don't necessarily eliminate very many lines of code, they help a ton with understandability and readability. I took me a decent amount of time to even remember what @params[:value] was referring to. But by representing specific logic with private methods that clearly state their function in their own names, that certainly helps out with that issue.